### ISSUE TITLE
`inconsistent_struct_constructor` doesn't work

### ISSUE THREAD

**Issue by amab8901** [rust-lang/rust-clippy#16152]
### Summary

was this lint even tested before PR got merged? It's not even a corner case. This is the most straightforward scenario I can think of when this lint is supposed to give a warning

### Lint Name

inconsistent_struct_constructor

### Reproducer

I tried this code:

```rust
fn main() {
    let structy = Structy {
        field2: "a".to_owned(),
        field9: "a".to_owned(),
        field6: "a".to_owned(),
        field5: "a".to_owned(),
        field7: "a".to_owned(),
        field8: "a".to_owned(),
        field4: "a".to_owned(),
        field1: "a".to_owned(),
        field3: "a".to_owned(),
    };
    dbg!(structy);
}

#[derive(Debug)]
pub struct Structy {
    pub field1: String,
    pub field2: String,
    pub field3: String,
    pub field4: String,
    pub field5: String,
    pub field6: String,
    pub field7: String,
    pub field8: String,
    pub field9: String,
}

```

I expected to see this happen:
warn that the instantiation is out of order

Instead, this happened:
silence

### Version

```text
rustc 1.91.0 (f8297e351 2025-10-28)
binary: rustc
commit-hash: f8297e351a40c1439a467bbbb6879088047f50b3
commit-date: 2025-10-28
host: x86_64-unknown-linux-gnu
release: 1.91.0
LLVM version: 21.1.2
```

**Comment by y21** [comment]
By default the lint only emits a warning if the struct expression doesn't have any initializers. So with the default configuration you would only get a warning for something like this:
```rs
#![warn(clippy::pedantic)]
fn main() {
    let field1 = "a".to_owned();
    let field2 = "a".to_owned();
    let structy = Structy {
        field2, // note that there is no initializer here
        field1
    };
    dbg!(structy);
}

#[derive(Debug)]
pub struct Structy {
    pub field1: String,
    pub field2: String,
}
```
If you want the lint to work with initializers too (what you're asking for) you need to set the `check-inconsistent-struct-field-initializers = true` option in your clippy.toml file. This isn't enabled by default since this is a bit stricter, changes semantics in general and sometimes requires more involved changes (as I mentioned in the last section of my comment here: https://github.com/rust-lang/rust-clippy/issues/16148#issuecomment-3586689308)

**Comment by amab8901** [comment]
this is really useful, thanks! ⭐
